docs/content/en/docs/ide_configuration.md

```diff
@@ -65,3 +67,33 @@ The main things that cause a rebuild are:
 Bazel compilation flags are set by default in the project root's `.bazelrc` in
 such a way as to avoid rebuilds during development as much as possible. This
 includes setting `-c dbg` and `--incompatible_strict_action_env`.
+
+### Pointing HEIR to a local clone of `llvm-project`
+
+Occasionally changes in HEIR will need to be made in tandem with upstream
+changes in MLIR. To facilitate this, you can tell `bazel` to use a local clone
+of `llvm-project` instead of a pinned commit hash.
+
+Do it as follows:
+
+```bash
+cat > bazel/import_llvm.bzl << EOF
+"""Provides the repository macro to import LLVM."""
+
+def import_llvm(name):
+    """Imports LLVM."""
+    native.new_local_repository(
+        name = name,
+        # this BUILD file is intentionally empty, because the LLVM project
+        # internally contains a set of bazel BUILD files overlaying the project.
+        build_file_content = "# empty",
+        path = "/path/to/llvm-project",
+    )
+EOF
+```
+
+This will require a full rebuild if the checked-out LLVM commit differs from
+the pinned commit hash in `bazel/import_llvm.bzl`.
```

Comment:
Can you expand a bit on the most efficient build workflow with this setup? If you want to run the MLIR tests/etc, do you have to build LLVM twice (once from the LLVM dir (using LLVM's CMake or bazel setup) and then a rebuild from HEIR as bazel sees the changed source files), or is there a way to avoid that (other than `ccache`)? E.g., can I run the mlir tests via HEIR's bazel setup, or is bazel smart enough to realize LLVM has already been built?

Reply:
The short answer is you would have to do everything through bazel. Added a commit with more details.